src/CPU/procedureCallOverhead.cpp

```diff
@@ -0,0 +1,183 @@
+// Experiment to measure overhead of procedure call
+
+#include <iostream>
+#include <chrono>
+#include <unistd.h>
+#include <sched.h>
+#include "timer.hpp"
+#include "stats.hpp"
+#include "proc.hpp"
+using namespace std;
+
+const int N_RUNS = 10;
+const int N_ITERATIONS = 1000;
+
+void noParam() {}
+
+void oneParam(int param1) {}
+
+void twoParams(int param1, int param2) {}
+
+void threeParams(int param1, int param2, int param3) {}
+
+void fourParams(int param1, int param2, int param3, int param4) {}
+
+void fiveParams(int param1, int param2, int param3, int param4,
+                 int param5) {}
+
+void sixParams(int param1, int param2, int param3, int param4, 
+                int param5, int param6) {}
+
+void sevenParams(int param1, int param2, int param3, int param4, 
+                  int param5, int param6, int param7) {}
+
+
+double noParamsCall() {
+    Timer t;
+    double time = 0.0;
+    for (int i = 0; i < N_ITERATIONS; i++) {
+        t.begin();
+        noParam();
+        t.end();
+        time += t.time_diff_micro();
+    }
+    return time / N_ITERATIONS;
+}
+
+double oneParamCall() {
+    Timer t;
+    double time = 0.0;
+    for (int i = 0; i < N_ITERATIONS; i++) {
+        t.begin();
+        oneParam(10);
+        t.end();
+        time += t.time_diff_micro();
+    }
+    return time / N_ITERATIONS;
+}
+
+double twoParamsCall() {
+    Timer t;
+    double time = 0.0;
+    for (int i = 0; i < N_ITERATIONS; i++) {
+        t.begin();
+        twoParams(10, 10);
```

Comment:
these calls are being made right, did you check if the compiler optimized them away?

Reply:
Yes, I checked by looking at the assembly code that the call was being made.
Also, the time taken by the call validates it.